pynamodb/attributes.py

```diff
@@ -947,10 +947,16 @@ def serialize(self, values):
         """
         rval = []
         for v in values:
-            attr_class = (self.element_type()
-                          if self.element_type
-                          else _get_class_for_serialize(v))
-            rval.append({attr_class.attr_type: attr_class.serialize(v)})
+            attr_class = self._get_serialize_class(v)
+            if self.element_type and not isinstance(attr_class, self.element_type):
+                raise ValueError("List elements must be of type: {}".format(self.element_type.__name__))
+            attr_type = attr_class.attr_type
+            attr_value = attr_class.serialize(v)
+            if self.element_type and attr_value is None:
```

Comment:
Why wouldn't we want to store `{"NULL": True}` if there's no explicit `element_type` specified?

Reply:
updated